### ISSUE TITLE
ability to disable logging output

### ISSUE THREAD

**Issue by markhaehnel** [obi1kenobi/cargo-semver-checks#590]
### Describe your use case

I want to use cargo-semver-checks as a library for my crate that bumps crate versions based on the public api of crate.
When running the check i don't want to have all the output.

Currently i can only set log levels from `error` to `trace`. But not to `off`.

### Describe the solution you'd like

An option on `Check` to disable logging completly.

### Additional Context

I'm running the check like this:

```rust
    let current = Rustdoc::from_root(path_to_project_root);
    let check = Check::new(current);
    let result = check.check_release(); // <-- causes the log output
```

I would also love to implement this when someone can give me a quick hint where and how this would fit in the current API design of cargo-semver-check.

**Comment by obi1kenobi** [comment]
Thanks for opening this issue.

The ability to use cargo-semver-checks as a library needs quite a bit of polish, and I'd love the help! We're missing at least a few things:
- The `CrateReport` type needs to say which lints found what and where. Right now via the library API, if you disable output, you'll only know that a crate failed to semver-check and nothing else.
- We have a lot of printing throughout that doesn't specify a level, and always prints (as you've observed). Almost all of those are `INFO/WARN/ERROR` level prints, but we need to determine which is which and then apply that. One way to do it is to make the `GlobalConfig::shell_print()` and related methods take a level.

Also, a word of warning: I don't think cargo-semver-checks (or any alternative) is suitable for _completely automatically_ bumping crate versions. In my opinion, there are still too many edge cases where that logic would do the wrong thing.

For example, right now cargo-semver-checks won't notice this change at all:
```rust
// Old
pub fn example(value: i64) {}

// New
pub fn example(value: String) {}
```

If you tried to automatically assign a version number via cargo-semver-checks here, it would suggest "patch." That's obviously wrong. We even have a section in the FAQ about this: https://github.com/obi1kenobi/cargo-semver-checks#will-cargo-semver-checks-catch-every-semver-violation

So if you decide to move forward with your tool, please prominently describe that limitation and link to that section in our FAQ, so that users don't get a nasty surprise.

**Comment by markhaehnel** [comment]
I know about the limitations, thank you for elaborating on it!

My plan is to have a hybrid approach. I will try to build a version from the git history using `git-cliff` and also building a one using `cargo-semver-checks`. In the end using the higher version of both to be on the safer side.

Sure there will not be 100% accuracy. But the tool should only be used in manual fashion to prepare a release. It will have some overrides to force a version bump on a specific semver level.

I will take a look at `cargo-semver-checks` code at the places you mentioned and will think about a good way to solve it and get back to you.

Have a nice weekend!

**Comment by obi1kenobi** [comment]
Implemented in #592.
